sources/packages/backend/apps/api/src/route-controllers/report/_tests_/e2e/report.institutions.controller.exportReport.e2e-spec.ts

```diff
@@ -0,0 +1,283 @@
+    programYear = await ensureProgramYearExists(db, PROGRAM_YEAR_PREFIX);
+  });
+
+  it("Should generate the offering details report when a report generation request is made with the appropriate program year and offering intensity.", async () => {
+    // Arrange
+
+    // Created 3 offerings as follows:
+    // 1st offering with 0 applications.
+    // 2nd offering with two applications for two different students belonging to the location of the institution for which the report is generated.
+    // 3rd offering with an application for a student belonging to the location of a different institution for which this report is not generated.
+    const auditUser = systemUsersService.systemUser;
+    const student = await saveFakeStudent(db.dataSource);
+    // 1st offering: with no submitted applications to it.
+    const firstFakeOffering = createFakeEducationProgramOffering(
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        auditUser,
+      },
+      { initialValues: { studyStartDate: "2010-09-01" } },
+    );
+    const firstSavedOffering = await db.educationProgramOffering.save(
+      firstFakeOffering,
+    );
+    // 2nd offering: with an application belonging to the location of the institution for which the report is generated.
+    const secondFakeOffering = createFakeEducationProgramOffering(
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        auditUser,
+      },
+      { initialValues: { studyStartDate: "2010-09-01" } },
+    );
+    const secondSavedOffering = await db.educationProgramOffering.save(
+      secondFakeOffering,
+    );
+    await saveFakeApplication(
+      db.dataSource,
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        student,
+        offering: secondSavedOffering,
+        programYear,
+      },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // 3rd offering: with an application belonging to the location of the institution for which the report is not generated.
+    const thirdFakeOffering = createFakeEducationProgramOffering(
+      {
+        institution: collegeC,
+        institutionLocation: collegeCLocation,
+        auditUser,
+      },
+      { initialValues: { studyStartDate: "2010-09-01" } },
+    );
+    const thirdSavedOffering = await db.educationProgramOffering.save(
+      thirdFakeOffering,
+    );
+    await saveFakeApplication(
+      db.dataSource,
+      {
+        institution: collegeC,
+        institutionLocation: collegeCLocation,
+        student,
+        offering: thirdSavedOffering,
+        programYear,
+      },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    const payload = {
+      reportName: "Offering_Details_Report",
+      params: {
+        offeringIntensity: {
+          "Full Time": true,
+          "Part Time": false,
+        },
+        programYear: 10,
```

Comment:
It is a gap from the previous PRs but I believe that the `programYear` is not validated in the API.
It must validate if it is one of the available ones in the dropdown after the dryRun is executed otherwise any valid number would be accepted.

Reply:
Please use the program year object as below.
```ts
 const payload = {
  reportName: "Offering_Details_Report",
  params: {
    offeringIntensity: {
      "Full Time": true,
      "Part Time": false,
    },
    programYear: programYear.id,
  },
};
```